### ISSUE TITLE
Impossible to select text in Edge or draw in Macromedia Freehand using stylus

### ISSUE THREAD

**Issue by WillAdams** [TheJoeFin/Windows10-Community#17]
Fall Creators Update and later change the behavior of the stylus so that one can't just directly select text with the stylus or draw in legacy applications --- instead the stylus is dumbed down to an 11th finger.

Please provide a user interface option to revert this back to the original behavior which dates back to Windows for Pen Computing.

**Comment by TheJoeFin** [comment]
What type of device are you using?

**Comment by gregordr** [comment]
> By executing the following from an elevated command line, the next time any legacy application starts it will get the prior pen behavior from the Windows 10 Anniversary Update: “reg add HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Pen /v LegacyPenInteractionModel /t REG_DWORD /d 1 /f”
> 
> Anytime you want to switch legacy applications back to the behavior introduced in the Windows 10 Fall Creator’s Update, execute this: “reg add HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Pen /v LegacyPenInteractionModel /t REG_DWORD /d 0 /f” 

 https://www.reddit.com/r/Windowsink/comments/8508fi/controlling_pen_behavior_in_windows_10/

Try this please

**Comment by jesstech** [comment]
There are multiple comments in that reddit post from others who have applied its registry hack workaround, who say that doing so breaks _other_ pen functionality (like pressure sensitivity in Photoshop).

From my understanding, this workaround only affects "legacy" applications, or applications configured to use the legacy tablet API.  It does nothing for applications that have been updated to implement the new pen behavior.  For example, Edge does not fire proper mousedown/mouseup DOM events when using the pen, so any javascript-based drawing app will instead receive touch events.

For this reason I have chosen to remain on Windows 10 build 1703 until a system setting is provided to _fully_ revert windows ink to pre-1709 behavior for all applications to treat pen input as a mouse.

A developer has provided a relevant bit of code to demonstrate how the value for pen input messages changed in build 1709+ here: https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/14277341/#comment-73

Thank you for your continued effort on a compromise.  We appreciate what the Windows Ink team has done so far on this issue, but the majority of creative professionals and pen enthusiasts want to be given an option to restore their purchase's functionality in its entirety.

**Comment by WillAdams** [comment]
Samsung Galaxy Book 12 with a Staedtler Noris Digital Stylus

**Comment by adamzea** [comment]
Microsoft removed the normal/good pen behavior and made it so you have to hold down the "right click" button on a pen in order to select text or manipulate content in UWP apps and the Edge web browser. The registry trick restores normal pen selection behavior in Win32 developed programs which is totally confusing for normal users.  Anyway, I think Microsoft does not realize that some pen-enabled Windows 10 tablet PCs do not actually have a right-click barrel button, thus making text selection and content manipulation impossible.  The Samsung Galaxy Book is one of them.  Yes, I agree that this is a major problem.

**Comment by WillAdams** [comment]
Also, the update makes it impossible to select drop-down menus in certain apps coded with the QT toolkit, notably Carbide Create and Carbide Motion

**Comment by jesstech** [comment]
So, is this issue ever going to get some attention?  It’s becoming more and more difficult to remain on 1703 to avoid this breaking change.

**Comment by TheJoeFin** [comment]
Good question. I don't know the answer to it though. This Repo is managed by the Windows community, so if you have any information you should share it here. Have you tried any of the Windows Insider builds? Do they make any changes to this issue?

**Comment by jesstech** [comment]
Not personally, no.  I’m still doing all I can to disable Windows Update after nearly being forced onto 1809 last week.

Word is newer Insider builds have added a button to Settings to essentially toggle the existing registry hack but do not restore full pen functionality.  Those of us affected by this issue haven’t been loud enough that this half-measure isn’t acceptable, likely because the majority of users affected at release were Wacom and not Surface users and cared primarily about the change breaking Photoshop and not the rest of Windows.

**Comment by WillAdams** [comment]
No, I've rolled back to 1703 twice now --- I bought this machine and the Staedtler Noris Digital Stylus together because they worked well, and as I expected when tried them out in the store --- Fall Creators Update and since breaks my machine, making it unusable. I need it to work, and I see no reason why I should be testing a system which Microsoft has intentionally broken, without offering a mechanism to revert it to work as it did when I purchased it.

If I'd realized that this was by design, rather than a bug, I'd've returned the machine after updating to Fall Creators Update broke it. http://forum.tabletpcreview.com/threads/fall-creators-update-1709-issues.73088/

**Comment by einarmagnus** [comment]
I'm having this problem now in shareX, a screenshot app. Selecting the screenshot area it misses the mark by about 30px. This is with a surface pro 6 and its pen. 

Perhaps there is a guide somewhere for developers on how to handle this broken API? The paint-applications seem to work fine. What are they doing differently?

**Comment by WillAdams** [comment]
So I guess the answer here is that Microsoft has chosen "consistency" with the lowest common denominator of Android stylus behavior and is going to allow Google to dictate interface concepts, ignoring decades of products going back to Windows for Pen Computing.

Could we please get a registry hack to get the traditional behavior back?

**Comment by TheJoeFin** [comment]
Is there any way to search for a reg key which might change stylus behavior?

**Comment by rice41** [comment]
let me know when you find solution for this. Im having only issue with blender 2.79 and for this ive installed linux mint. It kinda helps and its anoying.

**Comment by baglayan** [comment]
Still no fix aside from staying on an older version, what a shame

**Comment by WillAdams** [comment]
I recently bought a Samsung Galaxy Book 3 Pro 360 on the assumption that disabling Windows Ink would address this issue.

Unfortunately, it doesn't completely --- still can't select text in Edge (which of course, is now Google Chrome re-badged), but at least Freehand works (I'd've had to return it if it hadn't).

Does anyone know of a web browser for Windows 11 which allows selecting text using drag-select? _And_ which will allow browsing Discourse forums? Seamonkey and K-meleon work for the former, but not the latter.

**Comment by Ancient-Void** [comment]
You can in Firefox, but it's not perfect, and it requires an about:config change: dom.w3c_pointer_events.scroll_by_pen.enabled set it to False.
This allows for highlighting text with the pen (as it should be), but it still doesn't let you drag tabs or the like with the pen. You can with a double tap and hold, touch, but not pen (which makes no sense to me).

Edit:; just noticed the win11 requirement, not sure if it works on 11 (I'm still on a Surface Pro 3).

**Comment by WillAdams** [comment]
Thanks! Just tried it, worked perfectly, Firefox is now my browser of choice.

**Comment by Ancient-Void** [comment]
Update: found out about this config setting dom.w3c_pointer_events.dispatch_by_pointer_messages to FALSE and it seems to have made the pen work as expected within Firefox. Dragging tabs and other UI elements with the pen works, as does selecting text. I no longer feel like I need a mouse and a desk to use my tablet (at least in Firefox).

**Comment by WillAdams** [comment]
Annoyingly, yesterday's Windows 11 update toggled off "Let me use my pen as a mouse when it's available", so this stopped working, so I'm back to leaving Settings running at that page so I can turn that feature on/off depending on which application I am using....

**Comment by Ancient-Void** [comment]
That's really annoying, might be a Policy to fix that in place. I've found that I only need the second one (pointer messages) toggled, and it forces the pen to act as a mouse (mostly: I still get the handles when highlighting). But also, still on Win10.

**Comment by Ancient-Void** [comment]
Just thinking, you might be able to create a fairly simple batch or powershell script to toggle the setting and then add it to a shortcut or on screen macro pad like Tablet Pro or Tablet Friend.

**Comment by WillAdams** [comment]
Good point. 

Found: https://www.elevenforum.com/t/enable-or-disable-use-pen-as-mouse-in-windows-11.5690/

but I don't want to deal w/ the UAC thing, so I'll just make it a habit to re-launch Settings each time the machine reboots.
